kratos/sources/model_part.cpp

```diff
@@ -1584,6 +1582,15 @@ void ModelPart::SetBufferSize(ModelPart::IndexType NewBufferSize)
 
 }
 
+void ModelPart::SetBufferSizeSubModelParts(ModelPart::IndexType NewBufferSize)
```

Comment:
guys, this is NOT correct.

you can change the buffer size only of the root modelpart, and the behaviour is deliberate

current implementation is the following:

~~~cpp
void ModelPart::SetBufferSize(ModelPart::IndexType NewBufferSize)
{
    KRATOS_ERROR_IF(IsSubModelPart()) << "Calling the method of the sub model part "
        << Name() << " please call the one of the root model part: "
        << GetRootModelPart().Name() << std::endl;

    for(SubModelPartIterator i_sub_model_part = mSubModelParts.begin(); i_sub_model_part != mSubModelParts.end(); i_sub_model_part++)
    {
        i_sub_model_part->mBufferSize = NewBufferSize;
    }

    mBufferSize = NewBufferSize;

    auto nodes_begin = NodesBegin();
    const int nnodes = static_cast<int>(Nodes().size());
    #pragma omp parallel for firstprivate(nodes_begin,nnodes)
    for(int i = 0; i<nnodes; ++i)
    {
        auto node_iterator = nodes_begin + i;
        node_iterator->SetBufferSize(mBufferSize);
    }

}
~~~